scripts/helper-functions.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function github_checkout {
+  local REPO=$1
+  local VERSION=$2
+  local DIRNAME=$(basename "$1")
+
+  cd "${DEPENDENCY_DIR}"
+  if [ -z "${DIRNAME}" ]; then
+    echo "Failed to get repo name from $1"
+    exit 1
+  fi
+  if [ -d "${DIRNAME}" ] && prompt "${DIRNAME} already exists. Delete?"; then
+    rm -rf "${DIRNAME}"
+  fi
+  if [ ! -d "${DIRNAME}" ]; then
+    git clone -q "https://github.com/${REPO}.git"
+  fi
+  cd "${DIRNAME}"
+  git fetch -q
+  git checkout "${VERSION}"
+}
+
+function cmake_install {
+  local NAME=$(basename "$(pwd)")
+  local BINARY_DIR=_build
+  if [ -d "${BINARY_DIR}" ] && prompt "Do you want to rebuild ${NAME}?"; then
+    rm -rf "${BINARY_DIR}"
+  fi
+  mkdir -p "${BINARY_DIR}"
+
+  # CMAKE_POSITION_INDEPENDENT_CODE is required so that Velox can be built into dynamic libraries \
+  cmake -Wno-dev -B"${BINARY_DIR}" \
+    -GNinja \
+    -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
+    -DCMAKE_CXX_STANDARD=17 \
+    "${INSTALL_PREFIX+-DCMAKE_PREFIX_PATH=}${INSTALL_PREFIX-}" \
+    "${INSTALL_PREFIX+-DCMAKE_INSTALL_PREFIX=}${INSTALL_PREFIX-}" \
+    -DCMAKE_CXX_FLAGS="-mavx2 -mfma -mavx -mf16c -masm=intel -mlzcnt" \
+    -DBUILD_TESTING=OFF \
+    "$@"
+  ninja -C "${BINARY_DIR}" install
+}
```

Comment:
Got it.
As a side note, are new lines expected in every file? For example if I run cmake-format sometimes it removes new lines at the end in the C++ files. 
So since cmake-format was removing them,  I don't add them

Reply:
Ideally, add a new line. If cmake-format removes it, then leave it at that.